### PR TITLE
Validate external EFS allow traffic of cluster subnet CIDRs

### PR DESCRIPTION
**Pull request by chenwany**
### Description of changes
* If headnode and queues don't have additional custom securoty groups, the validator valiate there are at least an inbound rule and outbound rule of the security groups of EFS mount target that cover the CIDRs of headnode subnet and compute subnets

### Tests
* Unit test
* Manual test
   * EFS mount target only contains headnode subnet CIDR--> fail with validation
   * EFS mount target only contains compute subnet CIDR--> fail with validation
   * EFS mount target contains headnode and compute subnet CIDRs, without port 2049 --> fail with validation
   * EFS mount target contains headnode and compute subnet CIDRs, with port 2049 --> ok
### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/validators/cluster_validators.py [hunk 1]
```diff
@@ -680,6 +681,50 @@ def _validate(self, efs_id, avail_zones: set, are_all_security_groups_customized
...
+                        for ip_range in ip_ranges:
+                            if ip_network(ip_range.get("CidrIp")).supernet_of(subnet_cidr):
+                                out_access = True
+                                break
+
+            if not in_access or not out_access:
+                self._add_failure(
+                    "There is an existing Mount Target {0} in the Availability Zone {1} for EFS {2}, "
+                    "but it does not have a security group that allows inbound and outbound rules to allow traffics of "
+                    "subnet {3}. Please modify the Mount Target's security group, to allow traffic on subnet.".format(
```
Comment (gmarciani) [inline cli/src/pcluster/validators/cluster_validators.py:732]: typo: traffics => traffic

File: cli/src/pcluster/validators/cluster_validators.py [hunk 2]
```diff
@@ -680,6 +681,50 @@ def _validate(self, efs_id, avail_zones: set, are_all_security_groups_customized
...
+            in_access = False
+            out_access = False
+            for sec_group in security_groups_ids:
+                # Check all inbound rules
+                for rule in sec_group.get("IpPermissions"):
+                    if rule.get("IpRanges") or rule.get("PrefixListIds"):
+                        ip_ranges = rule.get("IpRanges")
+                        if ip_ranges:
+                            for ip_range in ip_ranges:
+                                if ip_network(ip_range.get("CidrIp")).supernet_of(subnet_cidr):
```
Comment (gmarciani) [inline cli/src/pcluster/validators/cluster_validators.py]: What about adding a unit test to cover this improvement?

**Comment by gmarciani** [comment]
May you please briefly describe the tests you made to validate these changes?

File: cli/src/pcluster/validators/cluster_validators.py (hunk 1 above)
Comment (chenwany) [reply cli/src/pcluster/validators/cluster_validators.py:732]: Fixed

File: cli/src/pcluster/validators/cluster_validators.py (hunk 2 above)
Comment (chenwany) [reply cli/src/pcluster/validators/cluster_validators.py]: Added

**Comment by gmarciani** [comment]
I think we should declare this as an enhancement in our changelog because we are actually changing the way we validate security groups.

**Review by gmarciani — DISMISSED** [review]

**Review by gmarciani — APPROVED** [review]

**Pull request merged**